packages/hierarchy-browser/src/utils/utils.ts

```diff
@@ -141,6 +145,35 @@ export function getStaticNeighborEntities(
 	}, [] as IEntityDetail[])
 }
 
+export function createEntityMap(
+	entities?: IEntityDetail[],
+): IEntityMap | undefined {
+	if (entities) {
+		return entities.reduce((acc, entity) => {
+			const id = entity.id
+			acc[id] = entity
+			return acc
+		}, {} as IEntityMap)
+	}
+}
+
+export function isEntitiesAsync(
+	entities?:
+		| (IEntityDetail[] | INeighborCommunityDetail[])
+		| ((
+				params: ILoadParams,
+		  ) => Promise<IHierarchyNeighborResponse | IHierarchyDataResponse>),
+): boolean {
+	return !Array.isArray(entities) || entities instanceof Promise
+}
+
+export function addLevelLabels(communities: ICommunityDetail[]): ICommunity[] {
+	const max = communities.length - 1
+	return communities.map((comm, index) =>
+		Object.assign({}, { ...comm, level: max - index }),
```

Comment:
Useless Object.assign. You create a new object when you use the splat syntax anyway. E.g. {...comm} is equivalent to Object.assign({}, comm}.  This can be simplified to just `{ ...comm, level: max - index }`